nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -117,6 +127,27 @@ def close_session(self):
             if self.session.state == "AUTH":
                 self.session.logout()
 
+    def test_authentication(self) -> Tuple[bool, str]:
+        """Method to validate the authentication of the Source.
+
+        Returns:
+            Tuple:
+                bool: True if authentication was successful, False otherwise
+                str: Message from authentication execution
+        """
+        try:
+            self.open_session()
+            self.close_session()
+            is_authenticated = True
+            message = "Test OK"
+        except Exception as exc:
+            is_authenticated = False
+            if isinstance(exc.args[0], bytes):
+                message = str(exc.args[0].decode())
```

Comment:
What's the scenario that this `if` condition is handling? I would have thought `str(exc)` would generally be the desired output?

Reply:
I was getting a `str` from `bytes` such as `b'ABC'`, so `decoding` first removes the `b''` (maybe there is a better way :) )